File: C_Programming/Competitive_Programming/CP_019.c
//////////////////////////////////////////////////////////////////
//
//  File Name    : Summation_Of_Non_Factors
//
//  Description  : This program accepts a number from the user
//                 and returns the summation of all its
//                 non-factors, excluding the number itself.
//
//  Example:
//      Input  : 12
//      Output : 50
//
//      Input  : 10
//      Output : 37
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

int SumNonFact(int iNo)
{
    int i = 0;

    int iAddNonFact = 0;

    for(i = 1; i < iNo; i++)
    {
        if(iNo % i != 0)
        {
            iAddNonFact = iAddNonFact + i;
        }
    }    
    return iAddNonFact;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter Number : \n");
    scanf("%d", &iValue);

    iRet = SumNonFact(iValue);

    printf("%d", iRet);

    return 0;
}

File: Java_Programming/Competitive_Programming/CP_019.java
//////////////////////////////////////////////////////////////////
//
//  File Name    : Find_Largest_Digit
//
//  Description  : This program accepts a number and
//                 finds the largest digit present
//                 in the given number.
//
//  Example:
//      Input  : 83429
//      Output : Largest Digit is : 9
//
//////////////////////////////////////////////////////////////////

class Logic
{
    void findLargestDigit(int iNum)
    {
        int iDigit = 0, iComp = 0;
        int iNo = iNum;

        while(iNo > 0)  
        {
            iDigit = iNo % 10; 

            if(iComp < iDigit)  
            {
                iComp = iDigit;  
            }
            iNo = iNo / 10;  
        }
        System.out.println("Largest Digit is : " + iComp);
    }
}

class CP_019
{
    public static void main(String A[])
    {
        Logic lobj = new Logic();

        lobj.findLargestDigit(83429);
    }
}